### ISSUE TITLE
Data spec doesn't produce same results as long form

### ISSUE THREAD

**Issue by markaddleman** [metosin/spec-tools#88]
I believe the following code should all conform the same way but, under spec-tools 0.5.1, but the spec created through data-spec fails.  
```
(:require 
            [clojure.spec.alpha :as s]
            [spec-tools.core :as st]
            [spec-tools.spec :as spec]
            [spec-tools.data-spec :as ds])

(let [conformer (st/type-conforming
                    (assoc
                      spec-tools.conform/json-type-conforming
                      :timestamp
                      (fn [a b]
                        (if-not (satisfies? Inst b)
                          (java.util.Date. b)
                          b))))]
    (s/def :test/timestamp (st/spec spec/inst? {:type :timestamp}))
    (s/def :test/ts :test/timestamp)
    (println (st/conform :test/timestamp
                         1508703225865
                         conformer))
    (println (st/conform (s/keys :req [:test/ts])
                         {:test/ts 1508703225865}
                         conformer))
    (println (st/conform (ds/spec :test {:test/ts :test/timestamp})
                         {:test/ts 1508703225865}
                         conformer))
    (println (st/explain-data (ds/spec :test {:test/ts :test/timestamp})
                              {:test/ts 1508703225865}
                              conformer)))
```

**Comment by miikka** [comment]
`ds/spec` does a bit too much of wrapping:

```clojure
(s/def :test/ts :test/timestamp)
(s/form :test/ts)
;;=> (spec-tools.core/spec {:spec spec-tools.spec/inst?, :type :timestamp})

(ds/spec :test {:test/ts :test/timestamp})
(s/form :test/ts)
;; => (spec-tools.core/spec {:spec (spec-tools.core/spec {:spec spec-tools.spec/inst?, :type :timestamp}), :type :date})
```

**Comment by miikka** [comment]
Likely not soon.

**Comment by ikitommi** [comment]
Should be fixed in `[metosin/spec-tools "0.6.0-20171221.064855-2"]`. Could you please verify?

**Comment by ikitommi** [comment]
Closing this as tested to work.
